osu.Desktop/Program.cs

```diff
@@ -69,23 +81,37 @@ public static int Main(string[] args)
                                 throw new TimeoutException(@"IPC took too long to send");
                         }
 
-                        return 0;
+                        return;
                     }
 
                     // we want to allow multiple instances to be started when in debug.
                     if (!DebugUtils.IsDebugBuild)
                     {
                         Logger.Log(@"osu! does not support multiple running instances.", LoggingTarget.Runtime, LogLevel.Error);
-                        return 0;
+                        return;
+                    }
+                }
+
+                if (host.IsPrimaryInstance)
+                {
+                    try
+                    {
+                        Logger.Log("Starting legacy IPC provider...");
+                        legacyIpc = new LegacyTcpIpcProvider();
+                        legacyIpc.MessageReceived += onLegacyIpcMessageReceived;
+                        legacyIpc.Bind();
+                        legacyIpc.StartAsync();
```

Comment:
This is likely a temporary location for this code just to see if this works as an MVP. Eventually I think this will be somewhere inside `OsuGame`?

Let me know if this change should be done immediately.

Reply:
Definitely shouldn't be on `OsuGame`. Has to be desktop-only.